dio/lib/src/options.dart

```diff
@@ -336,52 +343,55 @@ class _RequestConfig {
     this.method,
     this.receiveTimeout,
     this.sendTimeout,
-    Map<String, dynamic> extra,
-    Map<String, dynamic> headers,
-    String contentType,
+    Map<String, dynamic>? extra,
+    Map<String, dynamic>? headers,
+    String? contentType,
     this.responseType,
     this.validateStatus,
-    this.receiveDataWhenStatusError = true,
-    this.followRedirects = true,
-    this.maxRedirects = 5,
+    bool? receiveDataWhenStatusError,
```

Comment:
Here we can again use default parameters to simplify the code.
```suggestion
    this.receiveDataWhenStatusError = true,
```